### DISCUSSION TITLE
Is there something like InputScope for TextBox?

### DISCUSSION THREAD

**Discussion by AArnott** [AvaloniaUI/Avalonia#10484] [Q&A] [answered]
As I take my app to Android, I notice the software keyboard that pops up for TextBox doesn't support swipe input. It's as if the textbox hasn't told the keyboard to accept words. Further, I'd like the first letter to be capitalized. In WPF [a `TextBox.InputScope` property](https://learn.microsoft.com/uwp/api/windows.ui.xaml.controls.textbox.inputscope?view=winrt-22621) exists. I don't see anything like that for Avalonia though. 
Is there a way to tip off the keyboard as to the kind of input a TextBox is expecting in Avalonia?

**Comment by jp2masa** [comment, marked as answer]
https://github.com/AvaloniaUI/Avalonia/blob/91d881829c21ff46e049aeea1478f06729a9db60/src/Avalonia.Base/Input/TextInput/TextInputOptions.cs

**Reply by AArnott** [reply]
Thank you for the tip. I hadn't thought to look for an attached property from another class.

It took me a while to realize it was an attached property even with your link, so for the benefit of others, here is the answer:

```xaml
<TextBox Text="{Binding Name}" TextInputOptions.ContentType="Name" />
```

In my case, I was using a `DataGridTextColumn`, so I had to switch it to a template column:
```xaml
<DataGridTemplateColumn Header="{Binding PlayerNameColumnLabel}" Width="*">
    <DataGridTemplateColumn.CellTemplate>
        <DataTemplate>
            <TextBlock Text="{Binding Name}" Margin="5" />
        </DataTemplate>
    </DataGridTemplateColumn.CellTemplate>
    <DataGridTemplateColumn.CellEditingTemplate>
        <DataTemplate>
            <TextBox Text="{Binding Name}" TextInputOptions.ContentType="Name" />
        </DataTemplate>
    </DataGridTemplateColumn.CellEditingTemplate>
</DataGridTemplateColumn>
```

**Reply by AArnott** [reply]
This works in the emulator, but not on my Android phone (Galaxy S20 FE). It still has the generic 1-character at a time keyboard. Any idea why?

**Reply by jp2masa** [reply]
I tested an emulator and a real device, and both displayed the normal keyboard I think. Could you please post a screenshot of actual vs expected?

**Reply by AArnott** [reply]
The keyboard _looks_ the same in either case (for me, because the content type is Name which calls for a normal looking keyboard). The difference is that the default keyboard doesn't allow swipe input, and doesn't capitalize the first letter in each word, the way Name content type does. On the emulator, swipes and capitalization works, and on the device, it doesn't. I don't know how to take a screenshot that would capture that.

**Reply by jp2masa** [reply]
I see. For me, it has the same behavior on both: doesn't capitalize, but swipe input works fine. Still have to test latest nightly though, some things were fixed after latest preview IIRC.
